image/image.go

```diff
@@ -328,6 +328,20 @@ func makeChannelFromTrack(what, track, defaultChannel string) (string, error) {
 	return mch.Clean().String(), nil
 }
 
+// neededDefaultProviders returns the names of all default-providers for
+// the content plugs that the given snap.Info needs.
+func neededDefaultProviders(info *snap.Info) (cps []string) {
+	for _, plug := range info.Plugs {
+		if plug.Interface == "content" {
+			var dprovider string
+			if err := plug.Attr("default-provider", &dprovider); err == nil && dprovider != "" {
+				cps = append(cps, dprovider)
```

Comment:
Not super significant but perhaps map[string]bool instead of an array?

Reply:
Fair point, but since it's a bootstraping code and we're facing a couple of providers at most, I think it's fine. Unless we expect to use it more later in normal operations?